### PR TITLE
feat: ledger

### PR DESCRIPTION
**Pull request by bradleySuira**
- Added changes to client factory to seed ledger accounts

### CODE REVIEW

File: src/factory/client/clients.ts [hunk 1]
```diff
@@ -136,17 +133,25 @@ export function createEVMClient(
...
+        address: accountInfo.address
+      });
+    }
     walletProvider = new EvmLedgerProvider(
       {
         network: ethereumNetwork,
-        derivationPath,
+        derivationPath: accountInfo.derivationPath,
+        addressCache,
         transportCreator: ledgerTransportCreator,
```
Comment (monokh) [inline src/factory/client/clients.ts]: We can't asume that all clients will be able to support WebHID transport. 
This should be an option in `WalletOptions` similar to how ledger providers were handled before

File: src/factory/client/clients.ts [hunk 2]
```diff
@@ -91,19 +89,19 @@ export function createBtcClient(
...
       {
         network: bitcoinNetwork,
         addressType,
-        baseDerivationPath,
-        basePublicKey: account?.publicKey,
-        baseChainCode: account?.chainCode,
+        baseDerivationPath: accountInfo.derivationPath,
+        basePublicKey: accountInfo?.publicKey,
+        baseChainCode: accountInfo?.chainCode,
         transportCreator: ledgerTransportCreator,
```
Comment (monokh) [inline src/factory/client/clients.ts]: Same story as other comment. Can't assume this transport will work on mobile

**Review by monokh — CHANGES_REQUESTED** [review]

File: src/factory/client/clients.ts (hunk 1 above)
Comment (bradleySuira) [reply src/factory/client/clients.ts]: in this case seems like wallet core can handle using it and it's implemented inside the '@chainify/hw-ledger' in `WebHidTransportCreator` I will add it then to the wallet and send inside `WalletOptions` like you said

Comment (bradleySuira) [reply src/factory/client/clients.ts]: fixed

File: src/factory/client/clients.ts (hunk 2 above)
Comment (bradleySuira) [reply src/factory/client/clients.ts]: fixed

**Review by monokh — APPROVED** [review]

**Pull request merged**